door-runtime/src/jsMain/kotlin/com/ustadmobile/door/DatabaseBuilder.kt

```diff
@@ -1,64 +1,103 @@
 package com.ustadmobile.door
 
 import com.ustadmobile.door.ext.createInstance
+import com.ustadmobile.door.jdbc.SQLException
 import com.ustadmobile.door.migration.DoorMigration
+import com.ustadmobile.door.migration.DoorMigrationAsync
+import com.ustadmobile.door.migration.DoorMigrationStatementList
+import com.ustadmobile.door.migration.DoorMigrationSync
 import org.w3c.dom.Worker
-import wrappers.SQLiteDatasourceJs
-import kotlin.reflect.KClass
+import wrappers.*
 
-actual class DatabaseBuilder<T: DoorDatabase>(private var context: Any, private var dbClass: KClass<T>, private var dbName: String){
+actual class DatabaseBuilder<T: DoorDatabase>(private val builderOptions: DatabaseBuilderOptions,
+                                              private val dbExportCallback: DatabaseExportToIndexedDbCallback){
```

Comment:
Persistence should be triggered after changes within a certain timeout window by default.